src/app/carbon-estimation-table/carbon-estimation-table.component.html

```diff
@@ -0,0 +1,72 @@
+<!-- eslint-disable @angular-eslint/template/interactive-supports-focus -->
+<table role="treegrid" aria-label="Estimation of emissions" class="tce-table-fixed tce-w-full">
+  <thead>
+    <tr>
+      <th>Category</th>
+      <th class="tce-w-40">Emissions</th>
+    </tr>
+  </thead>
+  <tbody
+    (keydown.home)="homeEndKeyBoardEvent($event, true)"
+    (keydown.end)="homeEndKeyBoardEvent($event, false)"
+    (keydown.control.home)="homeEndKeyBoardEvent($event, true)"
+    (keydown.control.end)="homeEndKeyBoardEvent($event, false)"
+    (keydown.arrowup)="arrowKeyBoardEvent($event, 'up')"
+    (keydown.arrowdown)="arrowKeyBoardEvent($event, 'down')">
+    @for (tableItem of tableData(); track $index) {
+      @if (!tableItem.parent) {
+        <tr
+          role="row"
+          [tabindex]="$index === 0 ? 0 : -1"
+          [attr.aria-level]="tableItem.level"
+          [attr.aria-setsize]="tableItem.setSize"
+          [attr.aria-posinset]="tableItem.positionInSet"
+          [attr.aria-expanded]="tableItem.expanded"
+          (keydown.arrowright)="parentRowArrowKeyBoardEvent($event, 'right', tableItem.category, tableItem.expanded)"
+          (keydown.arrowleft)="parentRowArrowKeyBoardEvent($event, 'left', tableItem.category, tableItem.expanded)"
+          [style.background-color]="tableItem.colour.background">
+          <td
+            role="gridcell"
+            tabindex="-1"
+            class="tce-flex tce-text-white tce-items-center"
+            (keydown.enter)="toggle(tableItem.category)">
+            <!-- eslint-disable-next-line @angular-eslint/template/click-events-have-key-events -->
+            <div
+              tabindex="-1"
+              (click)="toggle(tableItem.category)"
+              class="material-icons-outlined hover:tce-opacity-50 tce-m-1 tce-cursor-pointer">
+              {{ tableItem.expanded ? 'expand_less' : 'expand_more' }}
+            </div>
```

Comment:
There are quite a lot of accessibility pitfalls related to not using `<button>` elements to do this sort of thing. I think we should have a single `<button>` element that toggles the expansion.

Reply:
I dont think the button is needed as it is following the [treegrid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/). So for keyboard users there are multiple ways to expand/collapse the row as part of the pattern which I think negates the potential pitfalls

Reply:
Ah ok, my bad. That's interesting. I still feel like it's strange to have one element for the mouse clicks and a separate one for the enter key and that the element that handles mouse clicks is not a button but I'll read the pattern more thoroughly.

Reply:
I think the div that handles the 'enter' key needs an aria label, at the moment the screen reader announces the 'expand_mode'/'expand_less' icon code which is ok but perhaps should be something like 'expand'/'collapse section'.

Reply:
Looking at the [example](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/) of the pattern, the arrow that handles the clicks isn't a button either and it doesn't seem you can navigate to it with the keyboard. The screen reader (NVDA) announces the cell entry (and the row and column numbers) when the cell that handles the 'enter' key is focussed rather than something like 'expand'/'collapse'. Their data is a little different to ours though.

Reply:
I was thinking of going to the other way and making the div aria-hidden as it is just for visual users and keyboard users would use the row action or enter on cell.

Reply:
Yes, that might be a good idea for the arrow icon. It's also currently announced when focusing the row as a whole but that doesn't have the 'enter' key expansion behaviour. Given that it only interacts with the mouse perhaps it's supposed to be hidden from screen readers. The arrow icons in the [W3C example](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/) seems to be hidden from NVDA.